docs/index.md

```diff
@@ -74,7 +74,7 @@ connection "hibp" {
 }
 ```
 
-- `api_key` - (required) The API key to access the HIBP API. Can also be set with the `HIBP_API_KEY` environment variable.
+- `api_key` - The API key to access the HIBP API. Can also be set with the `HIBP_API_KEY` environment variable. This is only required while querying `hibp_breached_account` and `hibp_paste` tables. However, you do not need an API key to query `hibp_breach` and `hibp_password` tables.
```

Comment:
```suggestion
- `api_key` - The API key to access the HIBP API. Can also be set with the `HIBP_API_KEY` environment variable. This is only required while querying `hibp_breached_account` and `hibp_paste` tables.
```

I'd leave out what tables don't need it, since it's more we'd have to maintain over time (even if the next 10 tables we add don't require it)